test/testdata/JKubeAppDeployment_Containerfile

```diff
@@ -0,0 +1,27 @@
+FROM registry.access.redhat.com/ubi9/openjdk-17:1.20-2.1726695177
+
+LABEL org.opencontainers.image.authors="CRCQE <devtools-crc-qe@redhat.com>"
+
+USER root
+
+# Install oc
+RUN curl -o oc.tar https://downloads-openshift-console.apps.sandbox-m4.g2pi.p1.openshiftapps.com/amd64/linux/oc.tar
+RUN tar -xf oc.tar
+RUN mv oc /usr/local/bin/oc
+
+# Install Git
+RUN microdnf install -y git
+
+USER default
+
+RUN git clone https://github.com/eclipse-jkube/jkube.git
+COPY <<-EOT /script.sh
+  # Given
+  oc login -u developer -p developer https://host.containers.internal:6443 --insecure-skip-tls-verify
+  oc new-project jkube-quarkus-app-deploy-flow-test
+  oc get pods
+  cd jkube/quickstarts/maven/quarkus
+  # When
+  mvn -U package oc:build oc:resource oc:apply
```

Comment:
So whole point of we are creating this containerfile is to clone the repo and have mvn plugin installed ?

Reply:
Eclipse JKube requires these things to be installed to work:
- Java 1.8+ Runtime (provided by `registry.access.redhat.com/ubi9/openjdk-17:1.20-2.1726695177`)
- Apache Maven/Gradle (provided by `registry.access.redhat.com/ubi9/openjdk-17:1.20-2.1726695177`)

We can move the clone part to the script. Or maybe don't clone any repository, I can add a simple Quarkus project here in test folder.

Reply:
> As per discussion in #4397 (comment) we though it's better to not add extra dependencies and do java based build inside a container.

Yes, that is what we want. Have we tried using this container file directly using oc command instead building it with podman and then running with podman? I think if we do it as part of oc then build happen in the cluster and we don't need the login part.

Reply:
Do you mean to execute it in a Pod instead?

Reply:
Thanks, Let me update this PR to use this approach instead.

Reply:
Can we keep it simple for now? The containerfile seems to be a situation that is easy to test with. The suggestion about building in a pod, and therefore involve S2I sounds like an improvement.

- The first step was to make sure the cluster is operational to run an application.
- The next step would be to make sure the expected functionality of the cluster is available/operational.
- The third step would be to load test, memory test, etc with an application we understand and 'own', to ensure the cluster keeps operating under conditions like stress, long use, etc.

Reply:
With current flow this looks like as part of containerfile we are embedding logic to login to cluster and then deploy it to cluster which seems overkill and have hardcoded cluster login info instead what we should've this containerfile generic enough to target any cluster.

Reply:
I have modified this PR to perform jkube build in a pod instead.